src/node.h

```diff
@@ -265,6 +265,9 @@ NODE_EXTERN Environment* CreateEnvironment(IsolateData* isolate_data,
 NODE_EXTERN void LoadEnvironment(Environment* env);
 NODE_EXTERN void FreeEnvironment(Environment* env);
 
+// This may return nullptr if context is not associated with a Node instance.
+NODE_EXTERN Environment* GetCurrentEnvironment(v8::Local<v8::Context> context);
```

Comment:
A question: What is the benefit of this without the `Environment` class declaration, which is internal?
https://github.com/nodejs/node/blob/1a2cf6696fa59c9723c423c4ff4c1167ab82155c/src/env.h#L25

Reply:
For one, you can test whether the current Context is a Node.js one or not, which is what electron does, I think.

Secondly, `Environment` would be the most reasonable target to improve the embedder API on, even if the class itself is not public. There are a couple of existing functions that take is as an argument, for example.

Reply:
The only caveat is that we need forward declarations:
https://github.com/nodejs/node/blob/0a90f943abe30a75078933f3b982de0096a4a569/src/node.h#L216-L219
(And they are exposed to internal code as well 🤷‍♂️)

Reply:
Now that I think of it, this might be way part of the reason I'm having trouble "embedded" node into node in https://github.com/nodejs/node/pull/23439.
I think that somehow we are violating the ODR...